socorro/unittest/processor/test_signature_utilities.py

```diff
@@ -1566,116 +1546,108 @@ def test_predicate_no_match(self):
         fake_processor = create_basic_fake_processor()
         rule = SigTrunc(fake_processor.config)
-        predicate_result = rule.predicate(rc, rd, pc, fake_processor)
+        predicate_result = rule.predicate({}, {}, processed_crash, fake_processor)
         ok_(predicate_result)
 
     def test_action_success(self):
-        pc = DotDict()
-        pc.signature = '9' * 256
-        rc = DotDict()
-        rd = {}
+        processed_crash = {
+            'signature': '9' * 256
+        }
         fake_processor = create_basic_fake_processor()
         rule = SigTrunc(fake_processor.config)
-        action_result = rule.action(rc, rd, pc, fake_processor)
+        action_result = rule.action({}, {}, processed_crash, fake_processor)
         ok_(action_result)
-        eq_(len(pc.signature), 255)
-        ok_(pc.signature.endswith('9...'))
+        eq_(len(processed_crash['signature']), 255)
+        ok_(processed_crash['signature'].endswith('9...'))
 
 
 class TestStackwalkerErrorSignatureRule(TestCase):
 
     def test_predicate_no_match(self):
-        pc = DotDict()
-        pc.signature = '0' * 100
-        rc = DotDict()
-        rd = {}
+        processed_crash = {
+            'signature': '0' * 100
+        }
         fake_processor = create_basic_fake_processor()
         rule = StackwalkerErrorSignatureRule(fake_processor.config)
-        predicate_result = rule.predicate(rc, rd, pc, fake_processor)
+        predicate_result = rule.predicate({}, {}, processed_crash, fake_processor)
         ok_(not predicate_result)
 
     def test_predicate(self):
-        pc = DotDict()
-        pc.signature = "EMPTY: like my soul"
-        rc = DotDict()
-        rd = {}
+        processed_crash = {
+            'signature': 'EMPTY: like my soul'
+        }
         fake_processor = create_basic_fake_processor()
         rule = StackwalkerErrorSignatureRule(fake_processor.config)
-        predicate_result = rule.predicate(rc, rd, pc, fake_processor)
+        predicate_result = rule.predicate({}, {}, processed_crash, fake_processor)
         ok_(predicate_result)
 
     def test_action_success(self):
-        pc = DotDict()
-        pc.signature = "EMPTY: like my soul"
-        pc.mdsw_status_string = 'catastrophic stackwalker failure'
-        rc = DotDict()
-        rd = {}
+        processed_crash = {
+            'signature': 'EMPTY: like my soul',
+            'mdsw_status_string': 'catastrophic stackwalker failure'
+        }
         fake_processor = create_basic_fake_processor()
         rule = StackwalkerErrorSignatureRule(fake_processor.config)
-        action_result = rule.action(rc, rd, pc, fake_processor)
+        action_result = rule.action({}, {}, processed_crash, fake_processor)
         ok_(action_result)
-        ok_(
-            pc.signature,
-            "EMPTY: like my soul; catastrophic stackwalker failure"
-        )
+        eq_(processed_crash['signature'], 'EMPTY: like my soul; catastrophic stackwalker failure')
```

Comment:
This was a bug. Instead of comparing two things to see if they were equal, it ended up checking to see if the first one was truthy with the second argument being what gets returned if it's not. I switched it from `ok_` to `eq_`.